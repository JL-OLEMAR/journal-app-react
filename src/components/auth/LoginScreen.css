.social-networks {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding-bottom: 20px;
  padding-top: 10px;
  width: 100%;

  & .btn-google {
    background-color: #4285f4;
    border-radius: 2px;
    box-shadow: 0 3px 4px 0 #00000025;
    cursor: pointer;
    height: 42px;
    width: 100%;
    transition: box-shadow 0.3s ease;

    & .google-icon-wrapper {
      background-color: #fff;
      border-radius: 2px;
      height: 40px;
      margin-left: 1px;
      margin-top: 1px;
      position: absolute;
      width: 40px;

      & .google-icon {
        height: 18px;
        margin-left: 11px;
        margin-top: 11px;
        position: absolute;
        width: 18px;
      }
    }

    & .btn-text {
      color: #fff;
      float: right;
      font-size: 14px;
      letter-spacing: 0.2px;
      margin: 11px 40px 0 0;
    }

    &:hover {
      box-shadow: 0 0 6px #4285f4;
    }

    &:active {
      background: #1669f2;
    }
  }
}
