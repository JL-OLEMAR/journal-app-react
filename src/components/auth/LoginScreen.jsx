import { useDispatch, useSelector } from 'react-redux'
import { Link } from 'react-router-dom'

import { startGoogleLogin, startLoginEmailPassword } from '../../actions/auth.jsx'
import { useForm } from '../../hooks/useForm.jsx'
import './LoginScreen.css'

export const LoginScreen = () => {
  const dispatch = useDispatch()
  const { loading } = useSelector(state => state.ui)

  const [formValues, handleInputChange] = useForm({
    email: 'test1@test.com',
    password: '123456'
  })

  const { email, password } = formValues

  const handleLogin = (e) => {
    e.preventDefault()
    dispatch(startLoginEmailPassword(email, password))
  }

  const handleGoogleLogin = () => dispatch(startGoogleLogin())

  return (
    <>
      <h3 className='auth-title'>Login</h3>
      <form
        className='animate__animated animate__fadeIn animate__faster'
        onSubmit={handleLogin}
      >
        <input
          autoComplete='off'
          className='auth-input'
          name='email'
          placeholder='Email'
          type='text'
          value={email}
          onChange={handleInputChange}
        />

        <input
          className='auth-input'
          name='password'
          placeholder='Password'
          type='password'
          value={password}
          onChange={handleInputChange}
        />

        <button
          className='btn btn-primary btn-block auth-btn'
          disabled={loading}
          type='submit'
        >
          Login
        </button>

        <div className='social-networks'>
          <p>or</p>
          <div
            className='btn-google'
            onClick={handleGoogleLogin}
          >
            <div className='google-icon-wrapper'>
              <img
                alt='google button'
                className='google-icon'
                src='https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg'
              />
            </div>
            <p className='btn-text'>
              <b>Sign in with google</b>
            </p>
          </div>
        </div>

        <Link
          className='link conf-center'
          to='/auth/register'
        >
          Create new account
        </Link>
      </form>
    </>
  )
}
