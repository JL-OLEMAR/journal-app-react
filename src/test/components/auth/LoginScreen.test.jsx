/**
 * @jest-environment node
 */

import { beforeEach, describe, expect, test } from 'vitest'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { mount } from 'enzyme'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import { LoginScreen } from '../../../components/auth/LoginScreen.jsx'
import 'jsdom-global/register' // → For mount a component

const middlewares = [thunk]
const mockStore = configureStore(middlewares)
const initState = { // → See in Redux DevTools
  auth: {},
  ui: {
    loading: false,
    msgError: null
  }
}
let store = mockStore(initState) // mock store

const wrapper = mount(
  <Provider store={store}>
    <MemoryRouter>
      <LoginScreen />
    </MemoryRouter>
  </Provider>
)

describe('Tests <LoginScreen />', () => {
  beforeEach(() => {
    store = mockStore(initState)
  })

  test('should wrapper the component LoginScreen', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
