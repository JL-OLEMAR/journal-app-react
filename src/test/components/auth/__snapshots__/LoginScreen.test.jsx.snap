// Vitest Snapshot v1

exports[`Tests <LoginScreen /> > should wrapper the component LoginScreen 1`] = `
[
  <h3
    className="auth-title"
  >
    Login
  </h3>,
  <form
    className="animate__animated animate__fadeIn animate__faster"
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      autoFocus={true}
      className="auth-input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="email"
      value=""
    />
    <input
      className="auth-input"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <button
      className="btn btn-primary btn-block auth-btn"
      disabled={false}
      type="submit"
    >
      Login
    </button>
    <div
      className="social-networks"
    >
      <p>
        or
      </p>
      <div
        className="btn-google mt-1"
        onClick={[Function]}
      >
        <div
          className="google-icon-wrapper"
        >
          <img
            alt="google button"
            className="google-icon"
            src="/src/assets/google-logo.svg"
          />
        </div>
        <p
          className="btn-text"
        >
          <b>
            Sign in with google
          </b>
        </p>
      </div>
    </div>
    <a
      className="link config-center"
      href="/auth/register"
      onClick={[Function]}
    >
      Create new account
    </a>
  </form>,
]
`;
